### PR TITLE
Using Simplify lambda expression and method reference syntax cleanup on

### PR DESCRIPTION
**Pull request by vogella**
core

Using the JDT UI "Simplify lambda expression and method reference syntax" clean-up on jdt.core.

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Review by akurtakov — APPROVED** [review]

File: org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java [hunk 1]
```diff
@@ -77,10 +77,8 @@ public TypeBinding resolveType(BlockScope scope) {
...
 		throw new CompletionNodeFound();
 	}
-
-	return new CompletionNodeFound(this, this.binding, scope).throwOrDeferAndReturn(() -> {
-		// probably not in the position to do useful resolution, just provide some binding
-		return this.resolvedType = new ProblemReferenceBinding(this.tokens, null, ProblemReasons.NotFound);
-	});
+	
+	// probably not in the position to do useful resolution, just provide some binding
+	return new CompletionNodeFound(this, this.binding, scope).throwOrDeferAndReturn(() -> (this.resolvedType = new ProblemReferenceBinding(this.tokens, null, ProblemReasons.NotFound)));
```
Comment (stephan-herrmann) [inline org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java:82]: This change is wrong. The code comment lost the connection to its code.
@vogella please don't mess up our code!

**Review by stephan-herrmann — COMMENTED** [review]
Did anyone really review this change?

File: org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java (hunk 1 above)
Comment (vogella) [reply org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java:82]: PR to move the comment back https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1695

Comment (stephan-herrmann) [reply org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java:82]: > PR to move the comment back #1695

No longer necessary, I had to recover it before I could fix #1440.

Comment (vogella) [reply org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java:82]: I leave it to you as experienced JDT developer to decide if you want to report that incorrect behaviour of the JDT clean-up action to jdt UI.

Comment (stephan-herrmann) [reply org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java:82]: > I leave it to you as experienced JDT developer to decide if you want to report that incorrect behaviour of the JDT clean-up action to jdt UI.

That looks like a perfect task for a person who loves these clean-up actions. I am not that person.

This incident only confirmed my worries that those actions may suggest that JDT is responsible for the outcome, whereas responsibility can only be owned by a person, viz. the contributor proposing the resulting change.

**Pull request merged**